core/src/main/java/com/sequenceiq/cloudbreak/converter/StackToTemplatePreparationObjectConverter.java

```diff
@@ -83,6 +91,10 @@ public TemplatePreparationObject convert(Stack source) {
             String stackRepoDetailsHdpVersion = hdpRepo != null ? hdpRepo.getHdpVersion() : null;
             Map<String, List<InstanceMetaData>> groupInstances = instanceGroupMetadataCollector.collectMetadata(source);
             String blueprintText = cluster.getBlueprint().getBlueprintText();
+            BlueprintStackInfo blueprintStackInfo = stackInfoService.blueprintStackInfo(blueprintText);
```

Comment:
Why didn't you change the code of `AmbariBlueprintViewProvider` to use the `BlueprintTextProcessorFactory`? It should be the `AmbariBlueprintViewProvider` responsibility to create the proper blueprint view.